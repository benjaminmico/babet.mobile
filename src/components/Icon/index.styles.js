import styled from 'styled-components'

export const IconContainer = styled.TouchableOpacity`
  width: 25px;
  height: 25px;
  justify-content: center;
  align-items: center;
`

export const EmptyView = styled.View``
