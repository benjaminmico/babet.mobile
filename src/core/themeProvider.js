import AsyncStorage from '@react-native-community/async-storage'
import React, {useContext, useEffect, useState} from 'react'
import {useDarkMode} from 'react-native-dark-mode'
import THEMES from './themes.json'

const STORAGE_KEY = 'THEME_ID'
const ThemeContext = React.createContext()

export const ThemeContextProvider = ({children}) => {
  const [themeID, setThemeID] = useState()

  const isDarkMode = useDarkMode ? useDarkMode() : false

  useEffect(() => {
    ;(async () => {
      // eslint-disable-next-line no-unused-expressions
      isDarkMode ? setThemeID(THEMES[0].key) : setThemeID(THEMES[1].key)
      // const storedThemeID = await AsyncStorage.getItem(STORAGE_KEY)
      // if (storedThemeID) setThemeID(storedThemeID)
      // // eslint-disable-next-line no-unused-expressions
      // else isDarkMode ? setThemeID(THEMES[0].key) : setThemeID(THEMES[1].key)
    })()
  }, [])

  return <ThemeContext.Provider value={{themeID, setThemeID}}>{themeID ? children : null}</ThemeContext.Provider>
}

export function withTheme(Component) {
  return props => {
    const {themeID, setThemeID} = useContext(ThemeContext)

    const getTheme = themeID => THEMES.find(theme => theme.key === themeID)
    const setTheme = themeID => {
      AsyncStorage.setItem(STORAGE_KEY, themeID)
      setThemeID(themeID)
    }

    return <Component {...props} themes={THEMES} theme={getTheme(themeID)} setTheme={setTheme} />
  }
}

export function getTheme() {
  const {themeID} = useContext(ThemeContext)

  const getThemeContent = themeID => THEMES.find(theme => theme.key === themeID)

  return getThemeContent(themeID)
}
