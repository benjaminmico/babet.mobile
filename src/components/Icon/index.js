/* eslint-disable global-require */
// @flow

import BankrollBlackIcon from '@assets/icons/blacks/bankrollIcon.svg'
import ContactUsBlackIcon from '@assets/icons/blacks/contactUsIcon.svg'
import GoBackBlackIcon from '@assets/icons/blacks/goBackIcon.svg'
import InformationsBlackIcon from '@assets/icons/blacks/informationsIcon.svg'
import LogOutBlackIcon from '@assets/icons/blacks/logOutIcon.svg'
import PasswordBlackIcon from '@assets/icons/blacks/passwordIcon.svg'
import PrivacyBlackIcon from '@assets/icons/blacks/privacyIcon.svg'
import SettingsBlackIcon from '@assets/icons/blacks/settingsIcon.svg'
import DeleteBlackIcon from '@assets/icons/blacks/deleteIcon.svg'
import DeleteWhiteIcon from '@assets/icons/whites/deleteIcon.svg'
import LostIcon from '@assets/icons/gameStatus/lost.svg'
import PendingIcon from '@assets/icons/gameStatus/pending.svg'
import WonIcon from '@assets/icons/gameStatus/won.svg'
import AddTicket from '@assets/icons/navigation/addTicket.svg'
import Profile from '@assets/icons/navigation/profile.svg'
import SelectedAddTicket from '@assets/icons/navigation/selectedAddTicket.svg'
import SelectedProfile from '@assets/icons/navigation/selectedProfile.svg'
import ShareIcon from '@assets/icons/share.svg'
import Tick from '@assets/icons/tick.svg'
import Ticket from '@assets/icons/ticket.svg'
import BankrollWhiteIcon from '@assets/icons/whites/bankrollIcon.svg'
import ContactUsWhiteIcon from '@assets/icons/whites/contactUsIcon.svg'
import GoBackWhiteIcon from '@assets/icons/whites/goBackIcon.svg'
import InformationsWhiteIcon from '@assets/icons/whites/informationsIcon.svg'
import LogOutWhiteIcon from '@assets/icons/whites/logOutIcon.svg'
import PasswordWhiteIcon from '@assets/icons/whites/passwordIcon.svg'
import PrivacyWhiteIcon from '@assets/icons/whites/privacyIcon.svg'
import SettingsWhiteIcon from '@assets/icons/whites/settingsIcon.svg'
import {withTheme} from '@core/themeProvider'
import React from 'react'
import {Image} from 'react-native'
import {EmptyView, IconContainer} from './index.styles'

type Props = {
  // icon name
  label: String,
  // icon size
  size: Number,
  // on press icon,
  onPress: () => void,
  // theme
  theme: Object,
}

/**
 *
 * @param {String} label icon name
 * @param {*} size icon size
 * @param {*} themeKey theme key to select color of the icon.
 * (SVG Tranformer => lib used to display svg do not have a color changing prop...)
 * @returns Icon to render
 */
const renderIcon = (label, size, themeKey) => {
  switch (label) {
    case 'tick':
      return <Tick width={size} height={size} />
    case 'share':
      return <ShareIcon width={size} height={size} />
    case 'informations':
      return themeKey === 'white' ? (
        <InformationsBlackIcon width={size} height={size} />
      ) : (
        <InformationsWhiteIcon width={size} height={size} />
      )
    case 'privacy':
      return themeKey === 'white' ? (
        <PrivacyBlackIcon width={size} height={size} />
      ) : (
        <PrivacyWhiteIcon width={size} height={size} />
      )
    case 'contact':
      return themeKey === 'white' ? (
        <ContactUsBlackIcon width={size} height={size} />
      ) : (
        <ContactUsWhiteIcon width={size} height={size} />
      )
    case 'update':
      return themeKey === 'white' ? (
        <ContactUsBlackIcon width={size} height={size} />
      ) : (
        <ContactUsWhiteIcon width={size} height={size} />
      )
    case 'delete':
      return themeKey === 'white' ? (
        <DeleteBlackIcon width={size} height={size} />
      ) : (
        <DeleteWhiteIcon width={size} height={size} />
      )
    case 'password':
      return themeKey === 'white' ? (
        <PasswordBlackIcon width={size} height={size} />
      ) : (
        <PasswordWhiteIcon width={size} height={size} />
      )
    case 'logOut':
      return themeKey === 'white' ? (
        <LogOutBlackIcon width={size} height={size} />
      ) : (
        <LogOutWhiteIcon width={size} height={size} />
      )
    case 'settings':
      return themeKey === 'white' ? (
        <SettingsBlackIcon width={size} height={size} />
      ) : (
        <SettingsWhiteIcon width={size} height={size} />
      )
    case 'bankroll':
      return themeKey === 'white' ? (
        <BankrollBlackIcon width={size} height={size} />
      ) : (
        <BankrollWhiteIcon width={size} height={size} />
      )
    case 'goBack':
      return themeKey === 'white' ? (
        <GoBackBlackIcon width={size} height={size} />
      ) : (
        <GoBackWhiteIcon width={size} height={size} />
      )
    case 'pending':
      return <PendingIcon width={size} height={size} />
    case 'won':
      return <WonIcon width={size} height={size} />
    case 'lost':
      return <LostIcon width={size} height={size} />
    case 'facebook':
      return (
        <Image
          style={{width: size, height: size, resizeMode: 'contain'}}
          source={require('../../../assets/icons/social/facebook.png')}
        />
      )
    case 'apple':
      return <Image style={{width: size, height: size}} source={require('../../../assets/icons/social/apple.png')} />
    case 'gmail':
      return <Image style={{width: size, height: size}} source={require('../../../assets/icons/social/gmail.png')} />
    case 'selectedProfile':
      return <SelectedProfile width={size} height={size} />
    case 'profile':
      return <Profile width={size} height={size} />
    case 'selectedAddTicket':
      return <SelectedAddTicket width={size} height={size} />
    case 'addTicket':
      return <AddTicket width={size} height={size} />
    case 'ticket':
      return <Ticket width={size} height={size} />
    default:
      return <EmptyView />
  }
}

const Icon = ({label, size, onPress, theme, ...props}: Props) => {
  // get theme props
  const {key} = theme

  // render icon
  return (
    <IconContainer disabled={!onPress} {...props} onPress={onPress}>
      {renderIcon(label, size, key)}
    </IconContainer>
  )
}

export default withTheme(Icon)
