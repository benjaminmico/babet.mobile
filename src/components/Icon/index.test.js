/**
 * @jest-environment jsdom
 */
import React from 'react'
import {StyleSheet} from 'react-native'
import {ThemeContextProvider} from '@core/themeProvider'
import {mount} from 'enzyme'
import Icon from '.'

describe.only('Icon', () => {
  describe('Integrity tests', () => {
    it('renders Icon correctly', () => {
      const label = 'bankroll'

      const wrapper = mount(
        <ThemeContextProvider>
          <Icon size={24} label={label} />
        </ThemeContextProvider>,
      )

      const IconView = wrapper.find('View').at(1)
      const IconItem = wrapper.find('SvgMock').first()

      expect(IconView.exists()).toBe(true)
      expect(IconItem.exists()).toBe(true)
    })
    it('renders Icon by missing entering label icon correctly', () => {
      const wrapper = mount(
        <ThemeContextProvider>
          <Icon size={24} />
        </ThemeContextProvider>,
      )

      const IconView = wrapper.find('View').at(1)
      const IconItem = wrapper.find('SvgMock').first()

      expect(IconView.exists()).toBe(true)
      expect(IconItem.exists()).toBe(false)
    })
  })

  describe('Props tests', () => {
    it('renders Icon', () => {
      const label = 'bankroll'

      const wrapper = mount(
        <ThemeContextProvider>
          <Icon size={24} label={label} />
        </ThemeContextProvider>,
      )

      const IconView = wrapper.find('View').first()

      expect(wrapper).toMatchSnapshot()

      const stylesContainer = StyleSheet.flatten(IconView.prop('style'))
      expect(stylesContainer.width).toBe(25)
      expect(stylesContainer.height).toBe(25)
      expect(stylesContainer.justifyContent).toBe('center')
      expect(stylesContainer.alignItems).toBe('center')
    })
  })
})
