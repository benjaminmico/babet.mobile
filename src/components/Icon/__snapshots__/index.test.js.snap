// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Props tests renders Icon 1`] = `
<ThemeContextProvider>
  <Component
    label="bankroll"
    size={24}
  >
    <Icon
      label="bankroll"
      setTheme={[Function]}
      size={24}
      theme={
        Object {
          "backgroundColor": "#ffffff",
          "color": "#101112",
          "colors": Object {
            "components": Object {
              "background": "#FFFFFF",
              "backgroundAction": "#F3F2F2",
              "backgroundContainer": "#FFFFFF",
              "backgroundFilter": "#F3F2F2",
              "backgroundPrimaryAction": "#5100FF",
              "icon": "#000000",
            },
            "palette": Object {
              "black": "#000000",
              "purple": "#5100FF",
              "white": "#FFFFFF",
            },
            "shadows": Object {},
            "texts": Object {
              "buttons": Object {
                "actions": "#5100FF",
                "cancellable": "#3E3E3E",
                "filter": "#3E3E3E",
                "isPrimaryAction": "#FFFFFF",
              },
              "description": "#ADADAD",
              "filters": Object {
                "multiFilter": "#5100FF",
                "selected": "#000000",
                "unselected": "#A3A2A2",
              },
              "input": Object {
                "label": "#7C7C7C",
                "value": "#000000",
              },
              "labelInput": "#7C7C7C",
              "text": "#000000",
              "title": "#000000",
            },
          },
          "isShadow": true,
          "key": "white",
          "statusBar": "dark-content",
        }
      }
      themes={
        Array [
          Object {
            "backgroundColor": "#101112",
            "color": "#ffffff",
            "colors": Object {
              "components": Object {
                "background": "#101112",
                "backgroundAction": "#F3F2F2",
                "backgroundContainer": "#2F3134",
                "backgroundFilter": "#F3F2F2",
                "backgroundPrimaryAction": "#5100FF",
                "icon": "#FFFFFF",
              },
              "palette": Object {
                "black": "#000000",
                "purple": "#5100FF",
                "white": "#FFFFFF",
              },
              "texts": Object {
                "buttons": Object {
                  "actions": "#5100FF",
                  "cancellable": "#3E3E3E",
                  "filter": "#3E3E3E",
                  "isPrimaryAction": "#FFFFFF",
                },
                "description": "#ADADAD",
                "filters": Object {
                  "multiFilter": "#5100FF",
                  "selected": "#000000",
                  "unselected": "#A3A2A2",
                },
                "input": Object {
                  "label": "#7C7C7C",
                  "value": "#FFFFFF",
                },
                "text": "#FFFFFF",
                "title": "#FFFFFF",
              },
            },
            "isShadow": false,
            "key": "dark",
            "statusBar": "light-content",
          },
          Object {
            "backgroundColor": "#ffffff",
            "color": "#101112",
            "colors": Object {
              "components": Object {
                "background": "#FFFFFF",
                "backgroundAction": "#F3F2F2",
                "backgroundContainer": "#FFFFFF",
                "backgroundFilter": "#F3F2F2",
                "backgroundPrimaryAction": "#5100FF",
                "icon": "#000000",
              },
              "palette": Object {
                "black": "#000000",
                "purple": "#5100FF",
                "white": "#FFFFFF",
              },
              "shadows": Object {},
              "texts": Object {
                "buttons": Object {
                  "actions": "#5100FF",
                  "cancellable": "#3E3E3E",
                  "filter": "#3E3E3E",
                  "isPrimaryAction": "#FFFFFF",
                },
                "description": "#ADADAD",
                "filters": Object {
                  "multiFilter": "#5100FF",
                  "selected": "#000000",
                  "unselected": "#A3A2A2",
                },
                "input": Object {
                  "label": "#7C7C7C",
                  "value": "#000000",
                },
                "labelInput": "#7C7C7C",
                "text": "#000000",
                "title": "#000000",
              },
            },
            "isShadow": true,
            "key": "white",
            "statusBar": "dark-content",
          },
        ]
      }
    >
      <Styled(Component)
        disabled={true}
        setTheme={[Function]}
        themes={
          Array [
            Object {
              "backgroundColor": "#101112",
              "color": "#ffffff",
              "colors": Object {
                "components": Object {
                  "background": "#101112",
                  "backgroundAction": "#F3F2F2",
                  "backgroundContainer": "#2F3134",
                  "backgroundFilter": "#F3F2F2",
                  "backgroundPrimaryAction": "#5100FF",
                  "icon": "#FFFFFF",
                },
                "palette": Object {
                  "black": "#000000",
                  "purple": "#5100FF",
                  "white": "#FFFFFF",
                },
                "texts": Object {
                  "buttons": Object {
                    "actions": "#5100FF",
                    "cancellable": "#3E3E3E",
                    "filter": "#3E3E3E",
                    "isPrimaryAction": "#FFFFFF",
                  },
                  "description": "#ADADAD",
                  "filters": Object {
                    "multiFilter": "#5100FF",
                    "selected": "#000000",
                    "unselected": "#A3A2A2",
                  },
                  "input": Object {
                    "label": "#7C7C7C",
                    "value": "#FFFFFF",
                  },
                  "text": "#FFFFFF",
                  "title": "#FFFFFF",
                },
              },
              "isShadow": false,
              "key": "dark",
              "statusBar": "light-content",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#101112",
              "colors": Object {
                "components": Object {
                  "background": "#FFFFFF",
                  "backgroundAction": "#F3F2F2",
                  "backgroundContainer": "#FFFFFF",
                  "backgroundFilter": "#F3F2F2",
                  "backgroundPrimaryAction": "#5100FF",
                  "icon": "#000000",
                },
                "palette": Object {
                  "black": "#000000",
                  "purple": "#5100FF",
                  "white": "#FFFFFF",
                },
                "shadows": Object {},
                "texts": Object {
                  "buttons": Object {
                    "actions": "#5100FF",
                    "cancellable": "#3E3E3E",
                    "filter": "#3E3E3E",
                    "isPrimaryAction": "#FFFFFF",
                  },
                  "description": "#ADADAD",
                  "filters": Object {
                    "multiFilter": "#5100FF",
                    "selected": "#000000",
                    "unselected": "#A3A2A2",
                  },
                  "input": Object {
                    "label": "#7C7C7C",
                    "value": "#000000",
                  },
                  "labelInput": "#7C7C7C",
                  "text": "#000000",
                  "title": "#000000",
                },
              },
              "isShadow": true,
              "key": "white",
              "statusBar": "dark-content",
            },
          ]
        }
      >
        <StyledNativeComponent
          disabled={true}
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "displayName": "Styled(Component)",
              "inlineStyle": InlineStyle {
                "rules": Array [
                  "
  width: 25px;
  height: 25px;
  justify-content: center;
  align-items: center;
",
                ],
              },
              "render": [Function],
              "styledComponentId": "StyledNativeComponent",
              "target": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          setTheme={[Function]}
          themes={
            Array [
              Object {
                "backgroundColor": "#101112",
                "color": "#ffffff",
                "colors": Object {
                  "components": Object {
                    "background": "#101112",
                    "backgroundAction": "#F3F2F2",
                    "backgroundContainer": "#2F3134",
                    "backgroundFilter": "#F3F2F2",
                    "backgroundPrimaryAction": "#5100FF",
                    "icon": "#FFFFFF",
                  },
                  "palette": Object {
                    "black": "#000000",
                    "purple": "#5100FF",
                    "white": "#FFFFFF",
                  },
                  "texts": Object {
                    "buttons": Object {
                      "actions": "#5100FF",
                      "cancellable": "#3E3E3E",
                      "filter": "#3E3E3E",
                      "isPrimaryAction": "#FFFFFF",
                    },
                    "description": "#ADADAD",
                    "filters": Object {
                      "multiFilter": "#5100FF",
                      "selected": "#000000",
                      "unselected": "#A3A2A2",
                    },
                    "input": Object {
                      "label": "#7C7C7C",
                      "value": "#FFFFFF",
                    },
                    "text": "#FFFFFF",
                    "title": "#FFFFFF",
                  },
                },
                "isShadow": false,
                "key": "dark",
                "statusBar": "light-content",
              },
              Object {
                "backgroundColor": "#ffffff",
                "color": "#101112",
                "colors": Object {
                  "components": Object {
                    "background": "#FFFFFF",
                    "backgroundAction": "#F3F2F2",
                    "backgroundContainer": "#FFFFFF",
                    "backgroundFilter": "#F3F2F2",
                    "backgroundPrimaryAction": "#5100FF",
                    "icon": "#000000",
                  },
                  "palette": Object {
                    "black": "#000000",
                    "purple": "#5100FF",
                    "white": "#FFFFFF",
                  },
                  "shadows": Object {},
                  "texts": Object {
                    "buttons": Object {
                      "actions": "#5100FF",
                      "cancellable": "#3E3E3E",
                      "filter": "#3E3E3E",
                      "isPrimaryAction": "#FFFFFF",
                    },
                    "description": "#ADADAD",
                    "filters": Object {
                      "multiFilter": "#5100FF",
                      "selected": "#000000",
                      "unselected": "#A3A2A2",
                    },
                    "input": Object {
                      "label": "#7C7C7C",
                      "value": "#000000",
                    },
                    "labelInput": "#7C7C7C",
                    "text": "#000000",
                    "title": "#000000",
                  },
                },
                "isShadow": true,
                "key": "white",
                "statusBar": "dark-content",
              },
            ]
          }
        >
          <ForwardRef
            disabled={true}
            setTheme={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 25,
                  "justifyContent": "center",
                  "width": 25,
                },
              ]
            }
            themes={
              Array [
                Object {
                  "backgroundColor": "#101112",
                  "color": "#ffffff",
                  "colors": Object {
                    "components": Object {
                      "background": "#101112",
                      "backgroundAction": "#F3F2F2",
                      "backgroundContainer": "#2F3134",
                      "backgroundFilter": "#F3F2F2",
                      "backgroundPrimaryAction": "#5100FF",
                      "icon": "#FFFFFF",
                    },
                    "palette": Object {
                      "black": "#000000",
                      "purple": "#5100FF",
                      "white": "#FFFFFF",
                    },
                    "texts": Object {
                      "buttons": Object {
                        "actions": "#5100FF",
                        "cancellable": "#3E3E3E",
                        "filter": "#3E3E3E",
                        "isPrimaryAction": "#FFFFFF",
                      },
                      "description": "#ADADAD",
                      "filters": Object {
                        "multiFilter": "#5100FF",
                        "selected": "#000000",
                        "unselected": "#A3A2A2",
                      },
                      "input": Object {
                        "label": "#7C7C7C",
                        "value": "#FFFFFF",
                      },
                      "text": "#FFFFFF",
                      "title": "#FFFFFF",
                    },
                  },
                  "isShadow": false,
                  "key": "dark",
                  "statusBar": "light-content",
                },
                Object {
                  "backgroundColor": "#ffffff",
                  "color": "#101112",
                  "colors": Object {
                    "components": Object {
                      "background": "#FFFFFF",
                      "backgroundAction": "#F3F2F2",
                      "backgroundContainer": "#FFFFFF",
                      "backgroundFilter": "#F3F2F2",
                      "backgroundPrimaryAction": "#5100FF",
                      "icon": "#000000",
                    },
                    "palette": Object {
                      "black": "#000000",
                      "purple": "#5100FF",
                      "white": "#FFFFFF",
                    },
                    "shadows": Object {},
                    "texts": Object {
                      "buttons": Object {
                        "actions": "#5100FF",
                        "cancellable": "#3E3E3E",
                        "filter": "#3E3E3E",
                        "isPrimaryAction": "#FFFFFF",
                      },
                      "description": "#ADADAD",
                      "filters": Object {
                        "multiFilter": "#5100FF",
                        "selected": "#000000",
                        "unselected": "#A3A2A2",
                      },
                      "input": Object {
                        "label": "#7C7C7C",
                        "value": "#000000",
                      },
                      "labelInput": "#7C7C7C",
                      "text": "#000000",
                      "title": "#000000",
                    },
                  },
                  "isShadow": true,
                  "key": "white",
                  "statusBar": "dark-content",
                },
              ]
            }
          >
            <TouchableOpacity
              disabled={true}
              hostRef={null}
              setTheme={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "height": 25,
                    "justifyContent": "center",
                    "width": 25,
                  },
                ]
              }
              themes={
                Array [
                  Object {
                    "backgroundColor": "#101112",
                    "color": "#ffffff",
                    "colors": Object {
                      "components": Object {
                        "background": "#101112",
                        "backgroundAction": "#F3F2F2",
                        "backgroundContainer": "#2F3134",
                        "backgroundFilter": "#F3F2F2",
                        "backgroundPrimaryAction": "#5100FF",
                        "icon": "#FFFFFF",
                      },
                      "palette": Object {
                        "black": "#000000",
                        "purple": "#5100FF",
                        "white": "#FFFFFF",
                      },
                      "texts": Object {
                        "buttons": Object {
                          "actions": "#5100FF",
                          "cancellable": "#3E3E3E",
                          "filter": "#3E3E3E",
                          "isPrimaryAction": "#FFFFFF",
                        },
                        "description": "#ADADAD",
                        "filters": Object {
                          "multiFilter": "#5100FF",
                          "selected": "#000000",
                          "unselected": "#A3A2A2",
                        },
                        "input": Object {
                          "label": "#7C7C7C",
                          "value": "#FFFFFF",
                        },
                        "text": "#FFFFFF",
                        "title": "#FFFFFF",
                      },
                    },
                    "isShadow": false,
                    "key": "dark",
                    "statusBar": "light-content",
                  },
                  Object {
                    "backgroundColor": "#ffffff",
                    "color": "#101112",
                    "colors": Object {
                      "components": Object {
                        "background": "#FFFFFF",
                        "backgroundAction": "#F3F2F2",
                        "backgroundContainer": "#FFFFFF",
                        "backgroundFilter": "#F3F2F2",
                        "backgroundPrimaryAction": "#5100FF",
                        "icon": "#000000",
                      },
                      "palette": Object {
                        "black": "#000000",
                        "purple": "#5100FF",
                        "white": "#FFFFFF",
                      },
                      "shadows": Object {},
                      "texts": Object {
                        "buttons": Object {
                          "actions": "#5100FF",
                          "cancellable": "#3E3E3E",
                          "filter": "#3E3E3E",
                          "isPrimaryAction": "#FFFFFF",
                        },
                        "description": "#ADADAD",
                        "filters": Object {
                          "multiFilter": "#5100FF",
                          "selected": "#000000",
                          "unselected": "#A3A2A2",
                        },
                        "input": Object {
                          "label": "#7C7C7C",
                          "value": "#000000",
                        },
                        "labelInput": "#7C7C7C",
                        "text": "#000000",
                        "title": "#000000",
                      },
                    },
                    "isShadow": true,
                    "key": "white",
                    "statusBar": "dark-content",
                  },
                ]
              }
            >
              <ForwardRef(AnimatedComponentWrapper)
                accessible={true}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Array [
                      Object {
                        "alignItems": "center",
                        "height": 25,
                        "justifyContent": "center",
                        "width": 25,
                      },
                    ],
                    Object {
                      "opacity": 1,
                    },
                  ]
                }
              >
                <AnimatedComponent
                  accessible={true}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Array [
                        Object {
                          "alignItems": "center",
                          "height": 25,
                          "justifyContent": "center",
                          "width": 25,
                        },
                      ],
                      Object {
                        "opacity": 1,
                      },
                    ]
                  }
                >
                  <View
                    accessible={true}
                    focusable={false}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "height": 25,
                        "justifyContent": "center",
                        "opacity": 1,
                        "width": 25,
                      }
                    }
                  >
                    <View
                      accessible={true}
                      focusable={false}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "height": 25,
                          "justifyContent": "center",
                          "opacity": 1,
                          "width": 25,
                        }
                      }
                    >
                      <SvgMock
                        height={24}
                        width={24}
                      />
                      <PressabilityDebugView
                        color="cyan"
                      />
                    </View>
                  </View>
                </AnimatedComponent>
              </ForwardRef(AnimatedComponentWrapper)>
            </TouchableOpacity>
          </ForwardRef>
        </StyledNativeComponent>
      </Styled(Component)>
    </Icon>
  </Component>
</ThemeContextProvider>
`;
